_data-prepper/pipelines/configuration/processors/add-entries.md

```diff
@@ -10,55 +10,215 @@ nav_order: 40
+```
+
+### Example: Add entries using value expressions
+
+The following example shows how you can configure the processor to use the `value_expression` option:
+
+```yaml
+... 
+  processor:
+    - add_entries:
+        entries:
+          - key: "length"
+            value_expression: "length(/message)"
+...
+```
+{% include copy.html %}
+
+When the input event contains the following data:
 
 ```json
 {"message": "hello"}
 ```
 
-And then you run the `add_entries` processor using the example pipeline, it adds a new entry, `{"newMessage": 3}`, to the existing event, `{"message": "hello"}`, so that the new event contains two entries in the final output:
+The processed event will contain the following data:
+
+```json
+{"message": "hello", "length": 5}
+```
+
+### Example: Add metadata
+
+The following example shows how you can configure the processor to add metadata to events:
+
+```yaml
+... 
+  processor:
+    - add_entries:
+        entries:
+          - metadata_key: "length"
+            value_expression: "length(/message)"
+...
+```
+{% include copy.html %}
+
+When the input event contains the following data:
 
 ```json
-{"message": "hello", "newMessage": 3}
+{"message": "hello"}
 ```
 
-If `newMessage` already exists, its existing value is overwritten with a value of `3`.
+The processed event will have the same data wit the metadata, `{"length": 5}`, attached. You can subsequently use expressions like `getMetadata("length")` in the pipeline. For more information, see the [`getMetadata` function](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/#getmetadata) documentation.
+
+
+### Example: Add dynamic key
 
+The following example shows how you can configure the processor to add metadata to events using a dynamic key:
+
+```yaml
+... 
+  processor:
+    - add_entries:
+        entries:
+          - key: "${/param_name}"
+            value_expression: "/param_value"
+...
+```
+{% include copy.html %}
+
+When the input event contains the following data:
+
+```json
+{"param_name": "cpu", "param_value": 50}
+```
+
+The processed event will contain the following data:
+
+```json
+{"param_name": "cpu", "param_value": 50, "cpu": 50}
+```
+
+### Example: Overwrite existing entries
+
+The following example shows how you can configure the processor to overwrite existing entries:
+
+```yaml
+... 
+  processor:
+    - add_entries:
+        entries:
+          - key: "message"
+            value: "bye"
+            overwrite_if_key_exists: true
+...
+```
+{% include copy.html %}
+
+When the input event contains the following data:
+
+```json
+{"message": "hello"}
+```
+
+The processed event will contain the following data:
+
+```json
+{"message": "bye"}
+```
+
+If `overwrite_if_key_exists` is not set to `true`, the input event will not be changed after going through the processor.
+
+### Example: Append to existing entries
+
+The following example shows how you can configure the processor to overwrite existing entries:
```

Comment:
It looks like this may be a copy and paste from the previous section. Should it read "The following example shows you how to configure the processor to append [noun] to existing entries"?

Reply:
Edited my suggestion.